app/src/main/java/com/goryachok/forecastapp/view/fragment/MyFragment.kt

```diff
@@ -1,10 +1,31 @@
 package com.goryachok.forecastapp.view.fragment
 
+import android.app.AlertDialog
+import android.location.Location
 import androidx.fragment.app.Fragment
+import com.goryachok.forecastapp.view.activity.MainActivity
+import com.goryachok.forecastapp.viewmodel.MyViewModel
 
-abstract class MyFragment: Fragment() {
+abstract class MyFragment(layout: Int) : Fragment(layout) {
+
+    abstract val viewModel: MyViewModel
 
     abstract fun onSearchRequest(request: String)
 
-    abstract fun onLocationRequest()
+    abstract fun onLocationRequest(loc: Location?)
+
+    override fun onResume() {
+        super.onResume()
+        val mainViewModel = activity?.let { (it as MainActivity).viewModel }
+        if (mainViewModel?.requestCache != "") {
```

Comment:
Please change to String.isNotEmpty()

Reply:
Changed to:
        if (!mainViewModel?.requestCache.isNullOrEmpty()) {